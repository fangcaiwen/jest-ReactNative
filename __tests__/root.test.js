/**
 * Created by 208439 on 2019/4/3
 *
 * Author: wind
 *
 * Content:root 测试用例
 */
import React from 'react';
import renderer from 'react-test-renderer';
import {configure, shallow,mount} from 'enzyme';
import Root from '../app/root';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


describe('Testing Root component', () => {

    // 测试UI
    it('root  renders correctly', () => {
        const tree = renderer.create(<Root />).toJSON();
        // 快照测试
        expect(tree).toMatchSnapshot();
    });

    // 测试逻辑
    it('renders Root as login expected', () => {
        const wrapper = shallow(
            <Root />
        );

        // 默认值没有登录
        expect(wrapper.state().isLogin).toBeFalsy();

        //设置登录值
        wrapper.setState({name:'11',psw:'222'});

        // 点击登录
        wrapper.props().children[4].props.callBack();

        // 登录成功
        expect(wrapper.state().isLogin).toBeTruthy();


    });
});
