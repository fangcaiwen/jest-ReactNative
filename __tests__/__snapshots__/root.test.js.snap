// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Root component root  renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCF8",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "left",
      }
    }
  >
    姓名：
  </Text>
  <View
    style={
      Object {
        "borderColor": "#888",
        "borderRadius": 4,
        "borderWidth": 1,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 10,
        "width": 300,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="请输入您所需要的字符"
      style={
        Object {
          "flex": 1,
          "fontSize": 16,
          "paddingLeft": 10,
        }
      }
      value=""
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "left",
      }
    }
  >
    密码：
  </Text>
  <View
    style={
      Object {
        "borderColor": "#888",
        "borderRadius": 4,
        "borderWidth": 1,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 10,
        "width": 300,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="请输入您所需要的字符"
      style={
        Object {
          "flex": 1,
          "fontSize": 16,
          "paddingLeft": 10,
        }
      }
      value=""
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 50,
        "opacity": 1,
        "width": 300,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00b7e3",
          "borderRadius": 4,
          "height": 50,
          "justifyContent": "center",
          "width": 300,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontSize": 18,
          }
        }
      >
        提交
      </Text>
    </View>
  </View>
</View>
`;
